src/parser/evaql/evaql_parser.g4

```diff
@@ -83,6 +84,10 @@ udfImpl
     : stringLiteral
     ;
 
+udfCost
+    : COST decimalLiteral
```

Comment:
Should the user provide the COST? Shouldn't it be something computed by the system and maintained internally? Can you provide an example where we might require the user to provide the COST?

Reply:
Ok, so my intuition is that if we use system profiled runtime as COST, it can depend on other factors such as input dimension. It is better to just ask user to input a relative COST? Otherwise, the monitored COST can change very frequently if input dimension changes.

Reply:
Even if the user provides the COST, it is dependent on the input. One more concern, shouldn't the cost be dependent on the underlying system (available hardware, memory, etc.)? I can understand the cost might fluctuate based on these parameters but even in the traditional databases, the cost estimates are not always accurate. We might have to keep adjusting based on the dataset (inputs), hardware, etc. 
With the current approach, we need to provide the user with guidelines on how the cost needs to be defined. And defining such generic guidelines will get complicated. We even have to handle scenarios where the user provides erroneous costs. The cost across UDFs have to be consistent, which might be a pain if multiple users are registering the UDFs. 

This is just my take on this. We should discuss this in our weekly meeting. @jarulraj @xzdandy your suggestions?